File: src/main/resources/modelsjs/Credit.js
const mongoose = require("mongoose");

const CreditSchema = new mongoose.Schema({
  description: {
    type: String,
    required: true,
    trim: true,
    maxlength: 100,
  },
  amount: {
    type: Number,
    required: true,
    min: 0.01,
  },
  dueDate: {
    type: Date,
    required: true,
  },
  creditor: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "BankAccount",
  },
  type: {
    type: String,
    enum: ["Loan", "Service Charge", "Credit Line"],
    required: true,
  },
  interestRate: {
    type: Number,
  },
  paymentMethod: {
    type: String,
    trim: true,
    maxlength: 50,
  },
  paid: {
    type: Boolean,
    default: false,
  },
  user: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
    ref: "Account",
  },
});

module.exports = mongoose.model("Credit", CreditSchema);

File: src/main/java/com/azaxxc/effintrakj/effinTrak/Credits/model/Credit.java
package com.azaxxc.effintrakj.effinTrak.Credits.model;

import com.azaxxc.effintrakj.effinTrak.accounts.model.BankAccount;
import com.azaxxc.effintrakj.effinTrak.users.models.User;
import jakarta.persistence.*;
import lombok.Data;
import java.util.Date;

@Data
@Entity
@Table(name = "credits")
public class Credit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String description;

    @Column(nullable = false)
    private Double amount;

    @Temporal(TemporalType.DATE)
    @Column(nullable = false)
    private Date dueDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creditor_id")
    private BankAccount creditor;

    @Column(nullable = false, length = 30)
    private String type;

    private Double interestRate;

    @Column(length = 50)
    private String paymentMethod;

    private Boolean paid = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

